modules/c++/math/source/Bessel.cpp

```diff
@@ -0,0 +1,175 @@
+/* =========================================================================
+ * This file is part of math-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2016, MDA Information Systems LLC
+ *
+ * math-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <cmath>
+#include <limits>
+
+#include <math/Bessel.h>
+
+namespace math
+{
+/*!
+ * Modified Bessel function of the first kind
+ * This function wraps the following order-specific functions
+ */
+double besselI(size_t order, double x)
+{
+    switch (order)
+    {
+        case 0:
+            return besselIOrderZero(x);
+
+        case 1:
+            return besselIOrderOne(x);
+
+        default:
+            return besselIOrderN(order, x);
+    }
+}
+
+/*!
+ * Modified Bessel function of the first kind, order 0
+ */
+double besselIOrderZero(double x)
+{
+    double ax = std::fabs(x);
```

Comment:
Fixed.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Tuesday, August 23, 2016 9:24 PM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [mdaus/coda-oss] Add Bessel function of the first kind (#158)


In modules/c++/math/source/Bessel.cpp<https://github.com/mdaus/coda-oss/pull/158#discussion_r75981484>:

> +            return besselIOrderZero(x);

> +

> +        case 1:

> +            return besselIOrderOne(x);

> +

> +        default:

> +            return besselIOrderN(order, x);

> +    }

> +}

> +

> +/*!

> + * Modified Bessel function of the first kind, order 0

> + */

> +double besselIOrderZero(double x)

> +{

> +    double ax = std::fabs(x);

Just use std::abs()

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mdaus/coda-oss/pull/158/files/02cc601d92da5a2e32d66734591e67287b7bc417#r75981484>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYLXQDwAOl2__UCicZmYQaPinBa25ks5qi508gaJpZM4JrCFu>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.